### ISSUE TITLE
Exception from Symphony on start if session already active

### ISSUE THREAD

**Issue by Gerdofal** [filegator/filegator#124]
**Describe the bug**
I've added a link to filegator from our menu, but running into a problem having to do with the session. We already have a session active and used to track the user, so when Filegator tries to load I get the following error:

Fatal error: Uncaught LogicException: Cannot change the name of an active session. in C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Proxy\AbstractProxy.php:117 Stack trace: #0 C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\NativeSessionStorage.php(201): Symfony\Component\HttpFoundation\Session\Storage\Proxy\AbstractProxy->setName('filegator') #1 C:\XFER\filegator\vendor\symfony\http-foundation\Session\Session.php(223): Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->setName('filegator') #2 C:\XFER\filegator\backend\Services\Session\Adapters\SessionStorage.php(35): Symfony\Component\HttpFoundation\Session\Session->setName('filegator') #3 C:\XFER\filegator\backend\App.php(33): Filegator\Services\Session\Adapters\SessionStorage->init(Array) #4 C:\XFER\filegator\dist\index.php(53): Filegator\App->__construct(Object(Filegator\Config\Config), Object(Filegator\Kernel\Request), Object(Filegator\Kernel\Response), Object(Filegator\Kernel\StreamedResponse), Object(Filegator in C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Proxy\AbstractProxy.php on line 117

**To Reproduce**
Happens on load

**Expected behavior**

**Screenshots**

**Logs**
Log contains only the error above

**Environment (please complete the following information):**
 
 - Server: IIS
 - PHP Version 7.2.14

**Additional context**

Ideally, we'd turn off sessions in FileGator. We don't want it to remember anything like last folder or anything else. Not sure that's an option though.

**Comment by alcalbg** [comment]
Hi,

Have you tried setting `$save_path` in your configuration.php file? This will store session files in a separate folder.

**Comment by Gerdofal** [comment]
Yes, tried that. Unfortunately, same error.

I also turned on warnings, here is the full output:


*Warning*: ini_set(): A session is active. You cannot change the session
module's ini settings at this time in
*C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Handler\NativeFileSessionHandler.php*
on
line *52*

*Warning*: ini_set(): A session is active. You cannot change the session
module's ini settings at this time in
*C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Handler\NativeFileSessionHandler.php*
on
line *53*

*Fatal error*: Uncaught LogicException: Cannot change the name of an active
session. in
C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Proxy\AbstractProxy.php:117
Stack trace: #0
C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\NativeSessionStorage.php(201):
Symfony\Component\HttpFoundation\Session\Storage\Proxy\AbstractProxy->setName('filegator')
#1
C:\XFER\filegator\vendor\symfony\http-foundation\Session\Session.php(223):
Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->setName('filegator')
#2
C:\XFER\filegator\backend\Services\Session\Adapters\SessionStorage.php(35):
Symfony\Component\HttpFoundation\Session\Session->setName('filegator') #3
C:\XFER\filegator\backend\App.php(33):
Filegator\Services\Session\Adapters\SessionStorage->init(Array) #4
C:\XFER\filegator\dist\index.php(53):
Filegator\App->__construct(Object(Filegator\Config\Config),
Object(Filegator\Kernel\Request), Object(Filegator\Kernel\Response),
Object(Filegator\Kernel\StreamedResponse), Object(Filegator in
*C:\XFER\filegator\vendor\symfony\http-foundation\Session\Storage\Proxy\AbstractProxy.php*
on
line *117*

**Comment by Gerdofal** [comment]
Is it possible to disable sessions, we don't need them.

**Comment by alcalbg** [comment]
Unfortunately you cannot disable session/cookies since they are used for authentication.

**Comment by Gerdofal** [comment]
That’s actually why we have a session created, because of authentication. Technically, I can close out that session before trying to share the files, but that would be kind of a pain. Let me see what I can figure out.

**Comment by Gerdofal** [comment]
Ok, maybe it would help if I told you what we're trying to do.

Basically, I'm embedding this into a tool we use for lots of other records from a database. Some have files related, and those files are stored on the server. Depending on which record the user has open, a different root folder should open.

I tried this first by changing the filesystem root folder in configuration.php based on a defined "ROOT_FOLDER_SELECTED"... but that failed because it looks like index.php reloads itself, and the second reload doesn't get the post data, so the definition is lost.

I tried storing the selected folder in the session we use for the main app, but that interferes with the session for Filegator....

I've tried a few other things too.  What it boils down to is I can't figure out a way to have a different root folder based on runtime information. I was hopeful about using the cookie, but that has too many conflicts.

Don't suppose you have a solution for this?

**Comment by Gerdofal** [comment]
```php 

//File user.php

public function getHomeDir(): string
    {
        if (!defined('HOME_DIRECTORY')) {
            throw new \Exception("Home directory is not defined.");
        }

        return HOME_DIRECTORY; //TODO - Make sure this is always defined
    }
```
Here is my ideal setup for what I'm trying to do. Unfortunately, I can't find a reliably consistent method of setting this at runtime without creating a fake "user" for every folder

**Comment by alcalbg** [comment]
Maybe you should look how the integration with filegator is done on this project: https://github.com/hestiacp/hestiacp

Basically, they have their own custom Auth and Session services: https://github.com/hestiacp/hestiacp/tree/main/install/deb/filemanager/filegator/backend/Services

**Comment by Gerdofal** [comment]
Thank you. This idea helped me a great deal. I have authentication handled and everything is working well except the homedir.

The problem is that setHomedir runs as expected on launch, I've added detailed logging to a file to make sure. It's set to the proper directory and everything is good. Then, everything clears and setHomedir is suddenly the root again because index.php is reloaded multiple times and the subsequent loads aren't referencing the information from the main menu. 

I'm thinking I messed up somewhere in the session maybe?

Here is a log of one load:

[2020-08-19 16:25:52] default.NOTICE Index.php loaded. Detected customDir (004223) passed in from main menu controller.
[2020-08-19 16:25:52] default.NOTICE customDir received in CustomAuth. Set to (004223)
[2020-08-19 16:25:52] default.NOTICE getGuest called. setHomedir to (004223)
[2020-08-19 16:25:52] default.NOTICE Index.php loaded. Detected customDir (null) passed in from main menu controller.
[2020-08-19 16:25:52] default.WARNING customDir not passed to CustomAuth
[2020-08-19 16:25:52] default.NOTICE getGuest called. setHomedir to ()
[2020-08-19 16:25:53] default.NOTICE Index.php loaded. Detected customDir (null) passed in from main menu controller.
[2020-08-19 16:25:53] default.WARNING customDir not passed to CustomAuth
[2020-08-19 16:25:53] default.NOTICE getGuest called. setHomedir to ()
[2020-08-19 16:25:53] default.NOTICE getGuest called. setHomedir to ()
[2020-08-19 16:25:53] default.NOTICE Index.php loaded. Detected customDir (null) passed in from main menu controller.
[2020-08-19 16:25:53] default.WARNING customDir not passed to CustomAuth
[2020-08-19 16:25:53] default.NOTICE getGuest called. setHomedir to ()
[2020-08-19 16:25:53] default.NOTICE getGuest called. setHomedir to ()
